src/lib/serviceRoutinesV2/postBatchQuery.cpp

```diff
@@ -62,6 +65,7 @@ std::string postBatchQuery
 {
   BatchQuery*           bqP  = &parseDataP->bq.res;
   QueryContextRequest*  qcrP = &parseDataP->qcr.res;
+  std::string           q    = ciP->uriParam[URI_PARAM_Q];
```

Comment:
"q" as URL param is only valid in "GET /v2/entities" operation (please check http://telefonicaid.github.io/fiware-orion/api/v2/)

...and in this case, note that is not a fix in a function preamble but in actual code.